driver/src/price_finding/stablex_naive_solver.rs

```diff
@@ -0,0 +1,208 @@
+use crate::price_finding::error::PriceFindingError;
+use crate::price_finding::PriceFinding;
+use crate::util;
+use dfusion_core::models::{AccountState, Order, Solution};
+use std::cmp::Ordering;
+
+/// A naive solver for the stableX contract. It works almost identically to the
+/// Snapp naive solver except it has a few extra requirements and it doesn't try
+/// to be as optimal.
+///
+/// The solver works by finding the first pair of orders which matching sell and
+/// buy tokens, where one of the traded tokens is the fee token, whose users have
+/// enough balance to fully fill, and that has a viable solution.
+///
+/// The solution is computed by taking the price of the larger order and and
+/// completely filling the smaller order. A large order is defined by having
+/// more non-fee token volume. Account balances are checked for each solution.
+pub struct StableXNaiveSolver;
+
+impl PriceFinding for StableXNaiveSolver {
+    fn find_prices(
+        &self,
+        orders: &[Order],
+        accounts: &AccountState,
+    ) -> Result<Solution, PriceFindingError> {
+        Ok(ipairs(orders)
```

Comment:
syntactic sugar. I like it!

Reply:
Why not use https://docs.rs/itertools/0.8.0/itertools/trait.Itertools.html#method.combinations ?

Reply:
I didn't want to pull in a whole crate just to enumerate pair indices. Especially since it is fairly easy to implement. Especially since this naive solver will eventually be merged with original one